src/GuzzleHttp/ClientSerializerAdapter.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/*
+ * This file is part of the CsaGuzzleBundle package
+ *
+ * (c) Charles Sarrazin <charles@sarraz.in>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code
+ */
+
+namespace Csa\Bundle\GuzzleBundle\GuzzleHttp;
+
+use GuzzleHttp\Client;
+use Psr\Http\Message\RequestInterface;
+
+class ClientSerializerAdapter extends Client
+{
+    /**
+     * @var string
+     */
+    private $jsonRegex = '(text/javascript)|(application/([^+]*\+){0,1}json)';
+
+    /**
+     * @var string
+     */
+    private $xmlRegex = 'application/([^+]*\+){0,1}xml';
+
+    /**
+     * @param RequestInterface $request
+     * @param array $options
+     *
+     * @return mixed
+     */
+    public function sendAsync(RequestInterface $request, array $options = [])
+    {
+        $options = $this->prepareDefaults($options);
+        $response = parent::sendAsync($request, $options)->wait();
+
+        return $this->processResponse($response, $options);
+    }
+
+    /**
+     * @param string $method
+     * @param null|string $uri
+     * @param array $options
+     *
+     * @return mixed
+     */
+    public function requestAsync($method, $uri = null, array $options = [])
```

Comment:
It is compatible with guzzle magic methods ?

It is compatible with magic methods